### PR TITLE
Convert volumes in [name:][host:]container[:access_mode] format

### PR DESCRIPTION
**Pull request by janetkuo**
#75

### CODE REVIEW

**Comment by ngtuna** [comment]
@janetkuo did you test with compose v1 which doesn't include volume name ?

**Comment by janetkuo** [comment]
Yes, I tested against `docker-gitlab.yml`, and an example output is:
```console
$ cat redisio-deployment.yaml 
apiVersion: extensions/v1beta1
kind: Deployment
metadata:
  creationTimestamp: null
  labels:
    service: redisio
  name: redisio
spec:
  replicas: 1
  selector:
    matchLabels:
      service: redisio
  strategy: {}
  template:
    metadata:
      creationTimestamp: null
      labels:
        service: redisio
    spec:
      containers:
      - image: sameersbn/redis
        name: redisio
        ports:
        - containerPort: 6379
          protocol: TCP
        resources: {}
        volumeMounts:
        - mountPath: /var/lib/redis
          name: ls6k1hh2gdnyxxvi7hvs
      restartPolicy: Always
      volumes:
      - hostPath:
          path: /srv/docker/gitlab/redis
        name: ls6k1hh2gdnyxxvi7hvs
status: {}
```

File: cli/app/app.go [hunk 1]
```diff
@@ -509,36 +509,70 @@ func configEnvs(name string, service ServiceConfig) []api.EnvVar {
...
-			volumeName := RandStringBytes(20)
+		name, host, container, mode, err := parseVolume(volume)
+		if err != nil {
+			logrus.Warningf("Failed to configure container volume: %v", err)
+			continue
+		}
 
-			volumesMount = append(volumesMount, api.VolumeMount{Name: volumeName, ReadOnly: readonly, MountPath: containerDir})
+		// volumeName = random string of 20 chars
+		if len(name) == 0 {
```
Comment (janetkuo) [inline cli/app/app.go]: I didn't assume that volume name is specified (even for v2, volume name isn't required), and if it's not, just use the current random volume name.

**Comment by janetkuo** [comment]
Just rebased.

**Comment by janetkuo** [comment]
I added a unit test, tested locally with `go test -tags experimental` (how do we enable testing in travis?)

File: cli/app/app.go (hunk 1 above)
Comment (ngtuna) [inline cli/app/app.go]: :+1:

**Pull request merged**